rmw_zenoh_cpp/src/detail/rmw_client_data.cpp

```diff
@@ -308,15 +231,14 @@ void ClientData::add_new_reply(std::unique_ptr<ZenohReply> reply)
     reply_queue_.size() >= adapted_qos_profile.depth)
   {
     // Log warning if message is discarded due to hitting the queue depth
-    z_view_string_t keystr;
-    z_keyexpr_as_view_string(z_loan(keyexpr_), &keystr);
+    std::string keystr = std::string(keyexpr_.value().as_string_view());
```

Comment:
In zenoh-cpp, the `string_view` can provide the string data with the corresponding length. It's no longer need to create another temp string.

Reply:
I'm getting this error 

```bash
conversion from ‘std::string_view’ {aka ‘std::basic_string_view<char>’} to non-scalar type ‘std::string’ {aka ‘std::__cxx11::basic_string<char>’} requested
  236 |     std::string keystr = keyexpr_.value().as_string_view();
```

Reply:
Answered in https://github.com/ros2/rmw_zenoh/pull/327#issuecomment-2536435747